### PR TITLE
updating linter version to stop segfaults on my machine

### PR DESCRIPTION
**Pull request by laudiacay**
the golang linter was segfaulting instantly on my machine. Updating its version fixed this.

### CODE REVIEW

**Comment by laudiacay** [comment]
fixes #1478

File: support/tools/go.mod [hunk 1]
```diff
@@ -1,8 +1,147 @@
 module github.com/filecoin-project/specs-actors/v6/support/tools
 
-go 1.16
+go 1.19
```
Comment (Stebalien) [inline support/tools/go.mod:3]: The current go version is 1.17. The next go 1.18. How are you running go 1.19?????

File: support/tools/go.mod [hunk 2]
```diff
@@ -1,8 +1,147 @@
...
 
 require (
 	github.com/Kubuxu/go-no-map-range v0.0.1
-	github.com/golangci/golangci-lint v1.28.2
+	github.com/golangci/golangci-lint v1.42.1
+)
+
+require (
```
[the remark this answers is not in the document]
Comment (Stebalien) [reply support/tools/go.mod:10]: Yeah, we shouldn't need a _second_ require statement. Does `go mod tidy` not clean this up?

File: support/tools/go.mod (hunk 1 above)
Comment (laudiacay) [reply support/tools/go.mod:3]: uhhh the jetlag caused me to time-travel forward to like 2022 (whoops! fixed it though, good catch)

File: support/tools/go.mod (hunk 2 above)
[the remark this answers is not in the document]
Comment (laudiacay) [reply support/tools/go.mod:10]: I don't think that changed the file but I'm going to commit a version where all of that is gone and see how CI feels about it

**Comment by Kubuxu** [comment]
We should change the `github.com/golangci/golangci-lint/pkg/lint` in `tools.go` to `github.com/golangci/golangci-lint/cmd/golangci-lint`.

The huge `require` block in go.mod is most likely expected due to the new Go's module pruning stuff.

**Review by ZenGround0 — APPROVED** [review]

**Pull request merged**